src/cargo/core/resolver/mod.rs

```diff
@@ -907,15 +907,22 @@ fn build_features<'a>(s: &'a Summary, method: &'a Method)
             }
             None => {
                 let feat = feat_or_package;
+
+                //if this feature has already been added, then just return Ok
                 if !visited.insert(feat) {
```

Comment:
There's `visited.remove(feat)` down the line, and it seems suspicious to me.

Can it lead to visiting and walking the same feature more then once? Does anything break if we just remove the `.remove`? I would think it was needed for better error reporting only.

Reply:
I will look into that later today.

Reply:
I agree. It makes sense to remove it. I've pushed a fix for that.